src/java/picard/analysis/CollectWgsMetrics.java

```diff
@@ -349,12 +358,18 @@ protected WgsMetrics generateWgsMetrics() {
         return new WgsMetrics();
     }
 
+    /**
+     * By design, if an INTERVALS file is provided, we want to count just those bases at the positions we care about,
+     * not across the entire read. Therefore, we call filter.getFilteredRecords() so that only the bases in the pileup
+     * at a given position are included in the calculations (with filter.getFilteredBases() we would be including other
+     * bases in the read too).
+     */
     protected long getBasesExcludedBy(final CountingFilter filter) {
-        return filter.getFilteredBases();
+        return (INTERVALS != null) ? filter.getFilteredRecords() : filter.getFilteredBases();
```

Comment:
Sorry if I am missing something, but aren't you using a `SamLocusIterator` that returns only records overlapping the intervals, and so you could just return `getFilteredBases` in either case?  What base would be counted that shouldn't?